sources/vega/commands/v1/transaction.proto

```diff
@@ -8,13 +8,23 @@ import "vega/commands/v1/validator_commands.proto";
 import "vega/commands/v1/oracles.proto";
 
 message InputData {
-  // A random number used to provided uniqueness and prevents against replay
-  // attack.
+  // A number to provide uniqueness, as to prevent accidental replays and,
+  // in combination with `block_height`, deliberate attacks.
+  // A hash over the transaction uniquely identifies the transaction, thus
+  // the nonce provides uniqueness for otherwise identical transactions. 
+  // Granted all other fields are equal, the nonce can either be a counter
+  // or generated at random to submit multiple transactions within the same
+  // block (see below), without being identified as replays.
+  // Please note that Protocol Buffers do not have a canonical, unique encoding
+  // and therefore different libraries or binaries may encode the same message
+  // slightly differently, causing a different hash.
   uint64 nonce = 1;
 
-  // The block height associated to the transaction.
-  // This should always be current height of the node at the time of sending the
-  // Tx. BlockHeight is used as a mechanism for replay protection.
+  // The block height at which the transaction is valid.
+  // This must be the current or a future block height, but never exceed the
+  // tolerance defined in the genesis state of the network (eg. 150 for mainnet1).
+  // Prefer to set this to the current block height.
+  // It prevents replay attacks in conjunction with `nonce` (see above).
```

Comment:
That is correct but also the network will not accept a height in the future. So you could use a future height if you want to prepare a transaction in advance, but not send it before it reach the accepted tolerance.

Reply:
Alright! What happens if the node you ask is slightly delayed? Is there a limit to transactions in a block? Is there any protection against getting arbitrary transactions dropped? In eth you can force the sequence of transactions through the counting nonce, and in btc you can force it through the graph of UTXOs you create

Reply:
well if the node is slightly delayed, then indeed it could be annoying, your transaction could be built with a block height reference to far away, which would get it rejected by the validators.
The inverse is also possible, your node could be part of the 2/3+1 validators, and be much more in advance from other nodes, and would eventually gives you a height in the future compared to slower nodes...

We do not have anything like btc or eth here unfortunately.